### ISSUE TITLE
eslint-disable-next-line doesn't work in style block of Vue single file component

### ISSUE THREAD

**Issue by GarrettGeorge** [eslint/eslint#11138]
**Tell us about your environment**

* **ESLint Version:** `4.15.0`
* **Node Version:** `v10.13.0`
* **npm Version:** `6.4.1`

**What parser (default, Babel-ESLint, etc.) are you using?** `Babel-ESLint` - version: `8.2.1`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint'
  },
  env: {
    browser: true,
  },
  // https://github.com/vuejs/eslint-plugin-vue#priority-a-essential-error-prevention
  // consider switching to `plugin:vue/strongly-recommended` or `plugin:vue/recommended` for stricter rules.
  extends: ['plugin:vue/essential', 'airbnb-base'],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  // check if imports actually resolve
  settings: {
    'import/resolver': {
      webpack: {
        config: 'build/webpack.base.conf.js'
      }
    }
  },
  // add your custom rules here
  rules: {
    'linebreak-style': 0,
    // don't require .vue extension when importing
    'import/extensions': ['error', 'always', {
      js: 'never',
      vue: 'never'
    }],
    // disallow reassignment of function parameters
    // disallow parameter object manipulation except for specific exclusions
    'no-param-reassign': ['error', {
      props: true,
      ignorePropertyModificationsFor: [
        'state', // for vuex state
        'acc', // for reduce accumulators
        'e' // for e.returnvalue
      ]
    }],
    // allow optionalDependencies
    'import/no-extraneous-dependencies': ['error', {
      optionalDependencies: ['test/unit/index.js']
    }],
    // allow debugger during development
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-underscore-dangle': ['error', { allow: ['_id'] }],
    'curly': ['error', 'multi-or-nest']
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// In Vue single file component, i.e. *.vue
<style lang="scss">
// eslint-disable-next-line
@import url('//fonts.googleapis....') // Line exceeds 100 characters
</style>
```

ESLint is run via the Webpack ESLint plugin

**What did you expect to happen?**

Expect client app to build without ESLint errors.

**What actually happened? Please include the actual, raw output from ESLint.**

```
WARNING in ./src/App.vue

  x https://eslint.org/docs/rules/max-len  Line 123 exceeds the maximum line length of 100
  src/App.vue:123:1
@import url('//fonts.googleapis....')
```

**Are you willing to submit a pull request to fix this bug?**

If it's a simple(ish) fix, sure.

**Comment by kaicataldo** [comment]
ESLint doesn't parse styles by default and the core rules are intended to be used with JavaScript only. Since it looks like [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue/) runs ESLint rules on styles, I suggest making an issue on that repository.

**Comment by mysticatea** [comment]
This is a limitation of `vue-eslint-parser` and `eslint-plugin-vue`.

As the [Known limitations section](https://github.com/mysticatea/vue-eslint-parser#%EF%B8%8F-known-limitations) mentioned, some core rules work with source code text rather than AST. This causes errors on outside of the `<script>` element.

`eslint-plugin-vue` recognize the `eslint-disable` comments in `<template>` and `<script>`. But it doesn't recognize the `eslint-disable` comments in the `<style>` element for now, because it doesn't have CSS parser. Unfortunately, there is the only workaround is to disable the rule by config file until the plugin supports CSS parsing.

I'm closing this issue because it's not a matter in ESLint itself.

**Comment by kaicataldo** [comment]
@mysticatea Thanks for the explanation :+1:
